implementations/rust/ockam/ockam_identity/tests/credentials_refresh.rs

```diff
@@ -0,0 +1,373 @@
+use std::sync::atomic::{AtomicU8, Ordering};
+use std::time::Duration;
+
+use ockam_core::async_trait;
+use ockam_core::compat::sync::Arc;
+use ockam_core::{route, Result};
+use ockam_identity::models::{CredentialAndPurposeKey, CredentialSchemaIdentifier};
+use ockam_identity::secure_channels::secure_channels;
+use ockam_identity::utils::AttributesBuilder;
+use ockam_identity::{
+    AuthorityService, Credentials, CredentialsRetriever, Identifier, IdentityError,
+    SecureChannelListenerOptions, SecureChannelOptions, TrustContext,
+};
+use ockam_node::Context;
+
+struct LocalCredentialsRetriever {
```

Comment:
My C++ background doesn't allow me to declare something after it's used :) I'll move it
But I don't see a point of having 2 implementations, looks like excess almost copy&paste

Reply:
There are already 2 implementations in the test code :-) ([L.49](https://github.com/build-trust/ockam/pull/6843/files#diff-3a42c5c069a583c53235d3eb1ddc5ed61966d334ce64ad9bcfda2667354e374eR49) and [L.105](https://github.com/build-trust/ockam/pull/6843/files#diff-3a42c5c069a583c53235d3eb1ddc5ed61966d334ce64ad9bcfda2667354e374eR105)).
My proposal is to wrap one implementation the other so that there's no duplication of behaviour:
```rust
struct FailingLocalCredentialsRetriever {
    credential_retriever: LocalCredentialsRetriever,
    fail_iteration: Option<u8>,
    failed_call_counter: Option<Arc<AtomicU8>>,
}
```

Reply:
@etorreborre the duplication was a mistake and is not removed :D